src/cascadia/TerminalSettingsEditor/Actions.cpp

```diff
@@ -4,61 +4,261 @@
+            //  key chord text box. It's hidden behind the data template.
+            //  Ideally, some kind of notification would alert the user, but
+            //  to make it look nice, we need it to somehow target the text box.
+        }
+        _RebindKeysRequestedHandlers(*this, *args);
+    }
+
     Actions::Actions()
     {
         InitializeComponent();
+    }
 
-        _filteredActions = winrt::single_threaded_observable_vector<Command>();
+    Automation::Peers::AutomationPeer Actions::OnCreateAutomationPeer()
+    {
+        _AutomationPeerAttached = true;
+        for (const auto& kbdVM : _KeyBindingList)
+        {
+            get_self<KeyBindingViewModel>(kbdVM)->IsAutomationPeerAttached(_AutomationPeerAttached);
+        }
+        return nullptr;
     }
 
     void Actions::OnNavigatedTo(const NavigationEventArgs& e)
     {
         _State = e.Parameter().as<Editor::ActionsPageNavigationState>();
 
-        std::vector<Command> keyBindingList;
-        for (const auto& [_, command] : _State.Settings().GlobalSettings().ActionMap().NameMap())
+        // Convert the key bindings from our settings into a view model representation
+        const auto& keyBindingMap{ _State.Settings().ActionMap().KeyBindings() };
+        std::vector<Editor::KeyBindingViewModel> keyBindingList;
+        keyBindingList.reserve(keyBindingMap.Size());
+        for (const auto& [keys, cmd] : keyBindingMap)
+        {
+            auto container{ make_self<KeyBindingViewModel>(keys, cmd) };
+            container->PropertyChanged({ this, &Actions::_ViewModelPropertyChangedHandler });
+            container->DeleteKeyBindingRequested({ this, &Actions::_ViewModelDeleteKeyBindingHandler });
+            container->RebindKeysRequested({ this, &Actions::_ViewModelRebindKeysHandler });
+            container->IsAutomationPeerAttached(_AutomationPeerAttached);
+            keyBindingList.push_back(*container);
+        }
+
+        std::sort(begin(keyBindingList), end(keyBindingList), KeyBindingViewModelComparator{});
+        _KeyBindingList = single_threaded_observable_vector(std::move(keyBindingList));
+    }
+
+    void Actions::_ViewModelPropertyChangedHandler(const IInspectable& sender, const Windows::UI::Xaml::Data::PropertyChangedEventArgs& args)
+    {
+        const auto senderVM{ sender.as<Editor::KeyBindingViewModel>() };
+        const auto propertyName{ args.PropertyName() };
+        if (propertyName == L"IsInEditMode")
         {
-            // Filter out nested commands, and commands that aren't bound to a
-            // key. This page is currently just for displaying the actions that
-            // _are_ bound to keys.
-            if (command.HasNestedCommands() || !command.Keys())
+            if (senderVM.IsInEditMode())
+            {
+                for (uint32_t i = 0; i < _KeyBindingList.Size(); ++i)
+                {
+                    const auto& kbdVM{ _KeyBindingList.GetAt(i) };

```

Comment:
```suggestion
                for (const auto& kbdVM : _KeyBindingList)
                {
```

right?

Reply:
oh never mind, we do need the index below

Reply:
Yeah. We _could_ use `ContainerFromItem`, but I figured `ContainerFromIndex` would be faster here since we're already iterating over all of the key binding view models anyways.